src/app/clusters/time-synchronization-server/TimeSyncDataProvider.cpp

```diff
@@ -0,0 +1,229 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "TimeSyncDataProvider.h"
+
+#include <lib/core/TLV.h>
+namespace chip {
+
+// Calculated with Python code:
+//   w = TLVWriter()
+//   dict([(0 , 0xff),(1, bytearray([0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff])), (2, 0xffff), ])
+//   w.put(None, s)
+//   len(w.encoding)
+constexpr size_t kTrustedTimeSourceMaxSerializedSize = 23u + 9u; // adding 9 as buffer
+constexpr size_t kTimeZoneMaxSerializedSize          = 90u + 6u; // adding 6 as buffer
+constexpr size_t kDSTOffsetMaxSerializedSize         = 34u + 6u; // adding 6 as buffer
```

Comment:
how did you decide between adding 9 or 6 as a buffer. A buffer for what?

Reply:
Good question, it is mostly guesstimation. It is mostly for safety in case the encoding introduces overhead. If there is a methodical way to do it, please let me know.

Reply:
`TLV::EstimateStructOverhead`?